events/reporter.go

```diff
@@ -198,15 +202,16 @@ func ReportError(err NodeError) {
 	defer mu.RUnlock()
 
 	if reporter != nil {
-		log.Info("about to report error updates for error : %v", err)
+		log.With().Info("about to report error updates for error", log.String("error-msg", err.Msg))
```

Comment:
```suggestion
		log.With().Info("about to report error updates for error", log.Err(err))
```

Reply:
Since `err` is of type `NodeError`, `log.Err(err)` does not work since the `NodeError` type does not have an `Error` method. Should I add the `Error` method, or leave this as is?